sw/device/lib/dif/dif_kmac_unittest.cc

```diff
@@ -150,7 +179,359 @@ class KmacTest : public testing::Test, public mock_mmio::MmioTest {
       setExpectedMessageByte(message, remaining);
     }
   }
+
+  void setExpectedConfig(void) {
+    EXPECT_WRITE32_SHADOWED(
+        KMAC_CFG_SHADOWED_REG_OFFSET,
+        {{KMAC_CFG_SHADOWED_KMAC_EN_BIT, config_.en},
+         {KMAC_CFG_SHADOWED_KSTRENGTH_OFFSET, config_.key_strength},
+         {KMAC_CFG_SHADOWED_MODE_OFFSET, config_.mode},
+         {KMAC_CFG_SHADOWED_MSG_ENDIANNESS_BIT, config_.msg_endianess},
+         {KMAC_CFG_SHADOWED_STATE_ENDIANNESS_BIT, config_.state_endianess},
+         {KMAC_CFG_SHADOWED_SIDELOAD_BIT, config_.sideload},
+         {KMAC_CFG_SHADOWED_ENTROPY_MODE_OFFSET, config_.entropy_mode},
+         {KMAC_CFG_SHADOWED_ENTROPY_FAST_PROCESS_BIT,
+          config_.entropy_fast_process},
+         {KMAC_CFG_SHADOWED_MSG_MASK_BIT, config_.msg_mask},
+         {KMAC_CFG_SHADOWED_ENTROPY_READY_BIT, config_.entropy_ready},
+         {KMAC_CFG_SHADOWED_ERR_PROCESSED_BIT, config_.err_processed},
+         {KMAC_CFG_SHADOWED_EN_UNSUPPORTED_MODESTRENGTH_BIT,
+          config_.en_unsupported_modestrength}});
+  }
+
+  void SetExpectedKey(const dif_kmac_key_t &key) {
+    std::map<dif_kmac_key_length_t, uint32_t> keySize = {
+        {kDifKmacKeyLen128, KMAC_KEY_LEN_LEN_VALUE_KEY128},
+        {kDifKmacKeyLen192, KMAC_KEY_LEN_LEN_VALUE_KEY192},
+        {kDifKmacKeyLen256, KMAC_KEY_LEN_LEN_VALUE_KEY256},
+        {kDifKmacKeyLen384, KMAC_KEY_LEN_LEN_VALUE_KEY384},
+        {kDifKmacKeyLen512, KMAC_KEY_LEN_LEN_VALUE_KEY512},
+    };
+
+    EXPECT_WRITE32(KMAC_KEY_LEN_REG_OFFSET, keySize[key.length]);
+    for (uint32_t i = 0; i < ARRAYSIZE(key.share0); ++i) {
+      ptrdiff_t offset = KMAC_KEY_SHARE0_0_REG_OFFSET + (i * sizeof(uint32_t));
```

Comment:
These parentheses are redundant; I would also inline `offset` into the `WRITE32` calls personally.